evap/staff/importers/json.py

```diff
@@ -0,0 +1,502 @@
+        else:
+            course_end = max(datetime.strptime(app["end"], self.DATETIME_FORMAT) for app in data["appointments"])
+
+        assert isinstance(data["isexam"], bool)
+        if data["isexam"]:
+            # Set evaluation time frame of three days for exam evaluations:
+            evaluation_start_datetime = course_end.replace(hour=8, minute=0, second=0, microsecond=0) + timedelta(
+                days=1
+            )
+            evaluation_end_date = (course_end + timedelta(days=3)).date()
+
+            name_de = data["title"].split(" - ")[-1] if " - " in data["title"] else "Prüfung"
+            name_en = data["title_en"].split(" - ")[-1] if " - " in data["title_en"] else "Exam"
+
+            weight = 1
+
+            # Update previously created main evaluation
+            # If events are graded for any program, wait for grade upload before publishing
+            if "courses" not in data or not data["courses"]:
+                wait_for_grade_upload_before_publishing = True
+            else:
+                wait_for_grade_upload_before_publishing = any(grade["scale"] for grade in data["courses"])
+            course.evaluations.all().update(
+                wait_for_grade_upload_before_publishing=wait_for_grade_upload_before_publishing
+            )
+        else:
+            # Set evaluation time frame of two weeks for normal evaluations:
+            # Start datetime is at 8:00 am on the monday in the week before the event ends
+            evaluation_start_datetime = course_end.replace(hour=8, minute=0, second=0, microsecond=0) - timedelta(
+                weeks=1, days=course_end.weekday()
+            )
+            # End date is on the sunday in the week the event ends
+            evaluation_end_date = (course_end + timedelta(days=6 - course_end.weekday())).date()
+
+            name_de, name_en = "", ""
+
+            weight = 9
+
+            # Might be overwritten when importing related exam evaluation
+            wait_for_grade_upload_before_publishing = True
+
+        participants = self._get_user_profiles(data["students"]) if "students" in data else []
+
+        defaults = {
+            "name_de": name_de,
+            "name_en": name_en,
+            "vote_start_datetime": evaluation_start_datetime,
+            "vote_end_date": evaluation_end_date,
+            "wait_for_grade_upload_before_publishing": wait_for_grade_upload_before_publishing,
+            "weight": weight,
+        }
+        evaluation, created = Evaluation.objects.get_or_create(
+            course=course,
+            cms_id=data["gguid"],
+            defaults=defaults,
+        )
+        if evaluation.state < Evaluation.State.APPROVED:
+            direct_changes = update_with_changes(evaluation, defaults)
+
+            participant_changes = set(evaluation.participants.all()) != set(participants)
+            evaluation.participants.set(participants)
```

Comment:
Should we use our helper here? Or is it wasteful when we don't actually care about the old values?